app.go

```diff
@@ -85,3 +88,11 @@ func Automatic(opts Options) *App {
 
 	return a
 }
+
+func (a *App) addDefaultRouteNames() {
```

Comment:
@markbates  Oh, yes i forget to talk about this, we need do it after because mux doesn't allow a route to name be changed after it was set ([look here](https://github.com/gorilla/mux/blob/34dda716af12ba79ed40ce1f66b102ff75dc3411/route.go#L115)).